lms/static/scripts/frontend_apps/config.ts

```diff
@@ -241,6 +248,13 @@ export type ConfigObject = {
 
   // Only present in "oauth2-redirect-error" mode.
   OAuth2RedirectError?: OAuthErrorConfig;
+
+  /**
+   * Change the current configuration after the application has initialized.
+   *
+   * Note that this property is installed by the frontend entry point.
+   */
+  update: (newConfig: Partial<ConfigObject>) => void;
```

Comment:
Perhaps it would make sense to split the "raw" definition of the config coming from the server and anything we set afterwards?

Or even handling this `update` "action" differently (I think I get the reason you did it this way, but it may not be super intuitive).

Reply:
> Perhaps it would make sense to split the "raw" definition of the config coming from the server and anything we set afterwards?

Yes, I think so. I added an `update` method here last week as it was the easiest way to give components the ability to update the config. It will probably be possible to find a nicer approach though.